### PR TITLE
fix anchor preview

### PR DESCRIPTION
**Pull request by jchamois**
Hello,

Here is a fix for the preview anchor. If a tag is present in the a tag, the preview of the anchor fail. You can reproduce by create an anchor in the editor and then put a bold on it via the toolbar. This way the 'b' tag is in the 'a' tag and the preview fails. 
I put a test on the target to check if we hover a a tag or a children.

Thanks

### CODE REVIEW

File: dist/js/medium-editor.js [hunk 1]
```diff
@@ -2468,13 +2468,24 @@ var AnchorPreview;
...
-        handleEditableMouseover: function (event) {
-            if (event.target && event.target.tagName.toLowerCase() === 'a') {
+       handleEditableMouseover: function (event) {
+            var target;
+
+             if(event.target){
+
+                if(event.target.tagName.toLowerCase() === 'a'){
+                    var target = event.target;
+                }else if(event.target.parentNode.tagName.toLowerCase() === 'a'){
```
Comment (j0k3r) [inline dist/js/medium-editor.js]: What if the text is bold **and** italic ?
Maybe you can take some inspiration from the [`setTargetBlank`](https://github.com/daviferreira/medium-editor/blob/master/src/js/util.js#L322-L333).

**Comment by j0k3r** [comment]
You should do your modification on `extension-anchor-preview.js` instead of on the dist file which is generated by Grunt

**Comment by nmielnik** [comment]
@jchamois , as @j0k3r pointed out, you should make your proposed changes in `extension-anchor-preview.js` and then run `grunt` to generate the dist files.  Without this your change will be wiped out the next time the dist files are generated.

**Comment by jchamois** [comment]
thanks for your comments, here is another commit. I check the closest a tag in order to allow to put italic, bold ad everything else.

**Comment by nmielnik** [comment]
@jchamois Hey, instead of making a change to the demo page, could you add a test to verify this change?  We probably don't want to change the demo as a proof that this works.

If you need any help, just let us know and we can try to help make a test.

**Comment by jchamois** [comment]
@nmielnik Hello ! I'va added test for my fix, I hope I've done it right ^^ Please let me know and thanks

**Comment by j0k3r** [comment]
Could you rebase your branch against the master? You'll have some conflicts to resolve.
Then, Travis will be able to launch the test suite.

File: src/js/extension-anchor-preview.js [hunk 2]
```diff
@@ -133,13 +134,36 @@ var AnchorPreview;
             this.instance_handleAnchorMouseout = null;
         },
 
+        getClosestTag : function(el, tag) {
```
Comment (j0k3r) [inline src/js/extension-anchor-preview.js]: I guess that function will need to be in the Util class instead.
Also, don't forget to add a comment above the function to tell what it should do.

**Comment by jchamois** [comment]
@j0k3r it's done ^^ Hope it's ok

**Comment by nmielnik** [comment]
@jchamois last things, could you revert the changes in `/demo/index.html`?  Since we have tests, it's best not to put this in the demo page.

**Comment by nmielnik** [comment]
Awesome, thanks @jchamois!

**Pull request merged**